app/models/challenge_models/gift_exchange.rb

```diff
@@ -17,26 +17,26 @@ class GiftExchange < ApplicationRecord
   belongs_to :potential_match_settings, dependent: :destroy
   accepts_nested_attributes_for :potential_match_settings
 
-  validates_length_of :signup_instructions_general, :signup_instructions_requests, :signup_instructions_offers, {
+  validates :signup_instructions_general, :signup_instructions_requests, :signup_instructions_offers, length: {
     allow_blank: true,
     maximum: ArchiveConfig.INFO_MAX, too_long: ts("must be less than %{max} letters long.", max: ArchiveConfig.INFO_MAX)
   }
+  validates :requests_num_required, numericality: { greater_than_or_equal_to: 1, only_integer: true }
+  validates :offers_num_required, numericality: { greater_than_or_equal_to: 1, only_integer: true }
```

Comment:
What's the advantage of adding these validations separately, instead of editing the validation below to have `greater_than_or_equal_to: 1`?

Reply:
To be honest, I missed that bit even after changing the style things :facepalm: If we wanted to maintain the current behavior of `_allowed` as 0 being corrected to the `_required` value, then I think the loop there would need to be split up; I'm not convinced that single case is worth the code duplication though. Thoughts?

Reply:
I think it makes sense to enforce that `_allowed` has to be at least 1, too. Letting it be 0 will only really affect things when `_required` is also 0 (otherwise it'll just get overridden in the `before_validation` hook), and I think an additional "Number of allowed requests per sign-up must be greater than or equal to 1." message wouldn't go awry in that case.